portfolio/src/main/java/com/google/sps/data/TagsUtils.java

```diff
@@ -27,11 +27,12 @@ public static void updateTagsToSearch (List<String> tagsToSearch) {
     if (!LoadFollowedTags.hasFollowedTags(email)) {
       return;
     }
-    List<String> newTags = LoadFollowedTags.getFollowedTags(email);
-    for (String tag : newTags) {
-      if (!tagsToSearch.contains(tag)) {
-        tagsToSearch.add(tag);
+    List<FollowedTag> userFollowedTags = LoadFollowedTags.getFollowedTags(email);
+    for (FollowedTag tag : userFollowedTags) {
+      if (!tagsToSearch.contains(tag.getTag())) {
```

Comment:
Can you leave a comment as to why we need to check for duplicate tags here? Does it make sense to use a hashset instead of iterating over the List for each tag?

Reply:
I think a hashset wouldn't be different here because the tags list received doesn't have duplicates. The duplicates could only be in datastore, and we don't want to put something in datastore that is already there.